src/component/modal/teamAddMemberModal/index.tsx

```diff
@@ -0,0 +1,88 @@
+import {
+  useState, useEffect, useCallback, memo, useMemo,
+} from 'react';
+import {
+  Typography, Box, List, TextField, ListItemText, Modal, ListItemButton,
+} from '@mui/material';
+import { useSearchMembers } from 'query/members';
+import { Member } from 'model/member';
+import { useCreateTeamMember } from 'query/teams';
+
+import * as S from './style';
+
+interface SearchMemberModalProps {
+  open: boolean;
+  onClose: () => void;
+  teamId: number;
+  isLeader: boolean;
+}
+
+const SearchModal = memo(({
+  open, onClose, teamId, isLeader,
+}: SearchMemberModalProps) => {
+  const [searchQuery, setSearchQuery] = useState('');
+  const [searchResults, setSearchResults] = useState<Member[]>([]);
+  const { mutate: addTeamMember } = useCreateTeamMember();
+
+  const { data: membersResult } = useSearchMembers({
+    pageIndex: 0, pageSize: 1000, name: searchQuery,
+  });
+
+  useEffect(() => {
+    if (membersResult) {
+      setSearchResults(membersResult.content);
+    } else {
+      setSearchResults([]);
+    }
+  }, [membersResult]);
+
+  const handleSearchChange = useCallback((e: any) => {
```

Comment:
TextField에서 변경되는 이벤트 타입은 `React.ChangeEvent<HTMLInputElement>`이므로 이걸 적용해주면 좋을 것 같아요
```suggestion
  const handleSearchChange = useCallback((e: React.ChangeEvent<HTMLInputElement>) => {
```